### PR TITLE
[fix] API Provider Data Type Mismatch #3807

### LINKED ISSUE #3807: API Provider Data Type Mismatch
See also **https://issues.jboss.org/browse/ENTESB-11568**

## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x ] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
When you provider Todo App swagger to API Provider a data type mismatch occurs on unimplemented flows:

image

image

### PR DESCRIPTION
**Pull request by KurtStam**
Fixes #3807, so we don't need a datamapper

### CODE REVIEW

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java [hunk 1]
```diff
@@ -162,8 +164,11 @@ private static ObjectNode createJsonSchemaWithProperties(final Map<String, Strin
         final List<AbstractSerializableParameter<?>> parameters = defaultProperties.entrySet().stream()//
```
Comment (zregvart) [inline app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java]: And I'd nuke this method too

**Comment by zregvart** [comment]
```
[INFO] There is 1 error reported by Checkstyle 8.10.1 with /syndesis/checkstyle.xml ruleset.
[ERROR] src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java:[65] (regexp) RegexpSingleline: Line has trailing spaces.
```

@KurtStam pre-commit hooks would take care of that for you, just sayin...

**Comment by zregvart** [comment]
Oh, and we need to follow this up with a change that includes the syndesis JSON schema URI in the data shape so that `ResponseCustomizer` can detect the need to unwrap the response here:

https://github.com/syndesisio/syndesis/blob/9e99679cd43adf3cfe171d4218ec4f31b2ea83a0/app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java#L64

That means we need to add `$id`[1] property to the unified JSON schemas. Similar to how the WebHook connector does it:

https://github.com/syndesisio/syndesis/blob/9e99679cd43adf3cfe171d4218ec4f31b2ea83a0/app/connector/webhook/src/main/java/io/syndesis/connector/webhook/WebhookConnectorCustomizer.java#L39

https://github.com/syndesisio/syndesis/blob/9e99679cd43adf3cfe171d4218ec4f31b2ea83a0/app/connector/webhook/src/main/java/io/syndesis/connector/webhook/WebhookConnectorCustomizer.java#L82

[1] https://json-schema.org/understanding-json-schema/structuring.html#the-id-property

**Comment by KurtStam** [comment]
@zregvart Adding the JSON Schema thingy seems like a different issue to me. Maybe we should open a new GH-issue for it, or am I missing something?

**Pull request merged**